builtin/logical/pki/acme_billing_test.go

```diff
@@ -0,0 +1,302 @@
+
+	// Using a different mount shouldn't affect counts.
+	doACMEForDomainWithDNS(t, dns, &acmeClientPKI2, []string{"dadgarcorp.com"})
+	expectedCount = validateClientCount(t, client, "", expectedCount, "different mount; same identifiers")
+
+	// But using a different identifier should.
+	doACMEForDomainWithDNS(t, dns, &acmeClientPKI2, []string{"pki2.dadgarcorp.com"})
+	expectedCount = validateClientCount(t, client, "pki2", expectedCount+1, "different mount with different identifiers")
+
+	// A new identifier in a unique namespace will affect results.
+	doACMEForDomainWithDNS(t, dns, &acmeClientPKINS1, []string{"unique.dadgarcorp.com"})
+	expectedCount = validateClientCount(t, client, "ns1/pki", expectedCount+1, "unique identifier in a namespace")
+
+	// But in a different namespace with the existing identifier will not.
+	doACMEForDomainWithDNS(t, dns, &acmeClientPKINS2, []string{"unique.dadgarcorp.com"})
+	expectedCount = validateClientCount(t, client, "", expectedCount, "existing identifier in a namespace")
+	doACMEForDomainWithDNS(t, dns, &acmeClientPKI2, []string{"unique.dadgarcorp.com"})
+	expectedCount = validateClientCount(t, client, "", expectedCount, "existing identifier outside of a namespace")
+
+	// Creating a unique identifier in a namespace with a mount with the
+	// same name as another namespace should increase counts as well.
+	doACMEForDomainWithDNS(t, dns, &acmeClientPKINS2, []string{"very-unique.dadgarcorp.com"})
+	expectedCount = validateClientCount(t, client, "ns2/pki", expectedCount+1, "unique identifier in a different namespace")
+}
+
+func validateClientCount(t *testing.T, client *api.Client, mount string, expected int64, message string) int64 {
+	resp, err := client.Logical().Read("/sys/internal/counters/activity/monthly")
+	require.NoError(t, err, "failed to fetch client count values")
+	t.Logf("got client count numbers: %v", resp)
+
+	require.NotNil(t, resp)
+	require.NotNil(t, resp.Data)
+	require.Contains(t, resp.Data, "non_entity_clients")
+	require.Contains(t, resp.Data, "months")
+
+	rawCount := resp.Data["non_entity_clients"].(json.Number)
+	count, err := rawCount.Int64()
+	require.NoError(t, err, "failed to parse number as int64: "+rawCount.String())
+
+	if expected != -1 {
+		require.Equal(t, expected, count, "value of client counts did not match expectations: "+message)
+	}
+
+	if mount == "" {
+		return count
+	}
+
+	/*mountResp, err := client.Logical().Read("sys/mounts/" + mount)
+	require.NoError(t, err, "failed to fetch mount entry information for " + mount)
+	t.Logf("got mount information: %v", mountResp)
+
+	require.NotNil(t, mountResp)
+	require.NotNil(t, mountResp.Data)
+	require.Contains(t, mountResp.Data, "accessor")
+	accessor := mountResp.Data["accessor"].(string)*/
+
+	months := resp.Data["months"].([]interface{})
+	if len(months) > 1 {
+		t.Skip("running across a month boundary; rerun test from start fully in the next month instead")
```

Comment:
we also have `timeutil.SkipAtEndOfMonth()` which will skip the test if the month is about to end.